WcaOnRails/lib/relations.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module Relations
+  def self.competitions_together(person1, person2)
+    Competition.find(person1.competition_ids & person2.competition_ids)
+  end
+
+  def self.compute_linkings
+    DbHelper.execute_sql File.read(Rails.root.join('lib', 'compute_linkings.sql'))
+  end
+
+  def self.get_chain(root_wca_id, other_wca_id)
+    left_tree = { degrees: { root_wca_id => 0 }, current_degree: 0 }
+    right_tree = { degrees: { other_wca_id => 0 }, current_degree: 0 }
+
+    wca_id_linking_trees = calculate_degrees_and_find_wca_id_linking_trees! left_tree, right_tree
+    return [] unless wca_id_linking_trees # Return an empty array if there is no realtion.
+    left_chain = build_chain root_wca_id, wca_id_linking_trees, left_tree[:degrees]
+    right_chain = build_chain other_wca_id, wca_id_linking_trees, right_tree[:degrees]
+    (left_chain + right_chain.reverse!).uniq!
```

Comment:
Is the call to `uniq!` just to get rid of the matching element in the middle? If so, could this be rewritten as this?

```ruby
left_chain[0...-1] + right_chain.reverse!
```